content_fetch/fetch_gc_articles_blog_post.js

```diff
@@ -1,57 +1,64 @@
-const fetch = require('node-fetch');
-const buildFileName = require("../utils/buildFileName");
-
-var getBlogPostsFromGCArticles = async function(lang) {
-  let url = lang == "en" ? process.env.GC_ARTICLES_ENDPOINT_EN + "posts?markdown=true&_embed" : process.env.GC_ARTICLES_ENDPOINT_FR + "posts?markdown=true&_embed"
-  return await fetch(url)
-  .then(response => response.json())
-  .then(
-    data => {
-      let files = [];
-      for (p in data) {
-        let post = data[p]
-        let replacedTitle = post.title.rendered.replace(/&#8217;/g, "’")
-        let category = post._embedded['wp:term']
-        let out = "";
-        out += "---\n";
-        out += "layout: blog\n";
-        out += "title: '" + replacedTitle + "'\n";
-        out += "description: >-\n";
-        out += "  " + post.markdown.excerpt.rendered + "\n";
-        out += "author: '" + post.meta.gc_author_name + "'\n";
-        out += "date: '" + post.date + "'\n";
-
-        out += "image: " + post._embedded['wp:featuredmedia'][0].media_details.sizes.full.source_url + "\n";
-        out += "image-alt: " + post._embedded['wp:featuredmedia'][0].alt_text + "\n";
-        out += "thumb: " + post._embedded['wp:featuredmedia'][0].media_details.sizes.full.source_url + "\n";
-
-        // if (category) {
-        //   let innerCategory = category[0]
-        //   let categoryArray = []
-
-        //   for (let i = 0; i < innerCategory.length; i++ ) {
-        //     categoryArray.push(`'${innerCategory[i].name}'`)
-        //   }
-
-        //   out += "category: " + `[${categoryArray}]` + "\n"
-        // } else {
-        //   out += "category: ['none']" + "\n"
-        // }
-        out += "translationKey: " + post.slug + "\n";
-        out += "---\n";
-
-        // Convert any body image URLS to Cloudfront
-
-        out += post.content.rendered + "\n";
-        
-        let slug = buildFileName(replacedTitle);
-
-        files.push({body: out, fileName: slug + ".md"})
-      }
-      
-      return files;
+const fetch = require( 'node-fetch' );
+const buildFileName = require( "../utils/buildFileName" );
+
+const generatePostContent = ( post ) => {
+  const replacedTitle = post.title.rendered.replace( /&#8217;/g, "'" );
+  const category = post._embedded[ 'wp:term' ];
+  
+  let out = '';
+  out += `---\n`;
+  out += `layout: blog\n`;
+  out += `title: '${ replacedTitle }'\n`;
+  out += `description: >-\n  '${ post.markdown.excerpt.rendered }'\n`;
+  out += `author: '${ post.meta.gc_author_name }'\n`;
+  out += `date: '${ post.date }'\n`;
+  out += `image: ${ post._embedded[ 'wp:featuredmedia' ][ 0 ].media_details.sizes.full.source_url }\n`;
+  out += `image-alt: ${ post._embedded[ 'wp:featuredmedia' ][ 0 ].alt_text }\n`;
+  out += `thumb: ${ post._embedded[ 'wp:featuredmedia' ][ 0 ].media_details.sizes.full.source_url }\n`;
```

Comment:
I had to make one change to check if there is actually an image on a given post, but otherwise it seems to be working as expected:
```suggestion
  if(post._embedded[ 'wp:featuredmedia' ]){
    out += `image: ${ post._embedded[ 'wp:featuredmedia' ][ 0 ].media_details.sizes.full.source_url }\n`;
    out += `image-alt: ${ post._embedded[ 'wp:featuredmedia' ][ 0 ].alt_text }\n`;
    out += `thumb: ${ post._embedded[ 'wp:featuredmedia' ][ 0 ].media_details.sizes.full.source_url }\n`;
  }
```